src/widgets/tables/merging_sets_screen.py

```diff
@@ -0,0 +1,409 @@
+
+        self.submit_button.setText("Submit")
+        self.submit_button.setEnabled(self.new_data is not None)
+        self.submit_button.clicked.connect(partial(self._click_listener, "submit"))
+        self.columns_merge_group_layout.addWidget(self.submit_button, 0)
+
+    def _click_listener(self, button_type: str):
+        match button_type:
+            case "load_file":
+                loading = LoadingWidget(self._load_from_file_handle)
+                loading.execute()
+            case "load_database":
+                loading = LoadingWidget(self._load_from_database_handle)
+                loading.execute()
+            case "submit":
+                loading = LoadingWidget(self._on_submit)
+                loading.execute()
+
+    def _load_from_file_handle(self):
+        self.import_state_label.setText("Loading ...")
+        file_path = QFileDialog.getOpenFileName(
+            self, "Choose file", ".", "*.csv *.json"
+        )[0]
+        try:
+            reader = self.loader.create_file_reader(file_path)
+        except ValueError as e:
+            self.import_state_label.setText(str(e))
+        else:
+            self._on_success(reader, file_path)
+
+    def _load_from_database_handle(self):
+        self.import_state_label.setText("Loading ...")
+        document_name = self.database_box.currentText()
+        try:
+            reader = self.loader.create_database_reader(document_name)
+        except ValueError as e:
+            self.import_state_label.setText(str(e))
+        else:
+            self._on_success(reader)
+
+    def _on_success(self, reader, file_path=None):
+        if file_path is not None:
+            self.filepath_line.setText(file_path)
+        self.import_state_label.clear()
+        if self.engine.is_data_big():
+            error = QMessageBox()
+            error.setIcon(QMessageBox.Warning)
+            error.setText("This file is too big.\nYou must save it in database!")
+            error.setWindowTitle("Warning")
+            error.exec_()
+        self.new_data = reader.read(None)
+        last_preview = self.layout.takeAt(2).widget()
+        if last_preview is not None:
+            last_preview.deleteLater()
+        self._render_table(self.new_data_view, False)
+
+        for i in reversed(range(self.columns_left_layout.count() - 1)):
+            # workaround to delete as low content as possible, removing nested layouts was giving me some segfault
+            self.columns_left_layout.itemAt(i).widget().setParent(None)
+
+        for i in reversed(range(self.columns_right_layout.count() - 1)):
+            self.columns_right_layout.itemAt(i).widget().setParent(None)
+
+        imported_columns = self._get_imported_columns()
+        for column in imported_columns:
+            self.columns_left_layout.insertWidget(
+                self.columns_left_layout.count() - 1,
+                ColumnWidget(column, self.columns_left_layout),
+            )
+
```

Comment:
```suggestion
        for i in reversed(range(self.columns_right_layout.count() - 1)):
            self.columns_right_layout.itemAt(i).widget().setParent(None)

```
Imported data doesn't change, so I think we can leave left layout as it is.